File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_cpp/p03383/p03383_num2_parsed.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

bool isSymmetric(vector<string>& grid, int H, int W) {
    for (int i = 0; i < H / 2; i++) {
        for (int j = 0; j < W / 2; j++) {
            if (grid[i][j] != grid[H - 1 - i][W - 1 - j]) {
                return false;
            }
        }
    }
    return true;
}

int main() {
    int H, W;
    cin >> H >> W;
    vector<string> grid(H);
    for (int i = 0; i < H; i++) {
        cin >> grid[i];
    }
    if (isSymmetric(grid, H, W)) {
        cout << "YES" << endl;
    } else {
        cout << "NO" << endl;
    }
    return 0;
}

File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_python/p03383/p03383_num2_parsed.py
H, W = map(int, input().split())
S = [input() for _ in range(H)]

def is_symmetric(S):
    for i in range(H//2):
        for j in range(W):
            if S[i][j] != S[H-1-i][W-1-j]:
                return False
    return True

if is_symmetric(S):
    print("YES")
else:
    print("NO")